### DISCUSSION TITLE
Web Searches for Intranet Sites

### DISCUSSION THREAD

**Discussion by ghost** [Flow-Launcher/Flow.Launcher#686] [Q&A]
Does the Web Searches plugin support intranet sites? For example, my workplace has two sites used for searching documentation (based on google). I configured the URL based on what I see in the URL bar when I make a search in the browser, but neither appear when using the defined Action Keyword

![image](https://user-images.githubusercontent.com/69704029/132543329-582891bf-0f4b-422b-b85c-491e171aa5f3.png)

As an example, Chrome can be configured to allow searching the intranet sites as so:

![image](https://user-images.githubusercontent.com/69704029/132542975-e746de39-c68b-40c5-bc9e-130c387464fb.png)

![image](https://user-images.githubusercontent.com/69704029/132542661-b78800c8-233c-4a95-bcb7-3302b330d44e.png)

**Comment by taooceros** [comment]
Could you upload the log? I think it should work.

**Comment by taooceros** [comment]
![image](https://user-images.githubusercontent.com/45326534/132606258-953afafb-a0a8-416b-9945-a27f56a263f8.png)
Probably you forget to enable it?

**Comment by ghost** [comment]
They are both enabled. I've attached the logs for your reference. Thank you for investigating. 
[2021-09-07.txt](https://github.com/Flow-Launcher/Flow.Launcher/files/7132693/2021-09-07.txt)
[2021-09-08.txt](https://github.com/Flow-Launcher/Flow.Launcher/files/7132694/2021-09-08.txt)

**Reply by taooceros** [reply]
No error is presenting in the log.....It works on my computer with the same link🤔

**Reply by taooceros** [reply]
Maybe some weird issue causing the websearch plugin unable to add the actionkeyword. Try adding "hub" and "guru" here. 
![image](https://user-images.githubusercontent.com/45326534/132607936-b34837d8-c25b-4acf-83fd-b1ac6521953c.png)
The files should locate at %appdata%\flowlauncher\settings\setting.json.

**Reply by ghost** [reply]
> No error is presenting in the log.....It works on my computer with the same link🤔

What do  you mean by this? Guru and Hubble are internal to my workplace. Do you work where I work?

**Reply by ghost** [reply]
Adding the actionKeywords "guru" and "hub" in _Settings.json_ didn't help, unfortunately.

**Reply by deefrawley** [reply]
Works for me too (we don't get through to the site dcavacec, but it opens a browser window with the correct URL and search term). So this is working for other plugin web searches? Can you add any other searches that work?

**Reply by jjw24** [reply]
> Maybe some weird issue causing the websearch plugin unable to add the actionkeyword. Try adding "hub" and "guru" here.
> ![image](https://user-images.githubusercontent.com/45326534/132607936-b34837d8-c25b-4acf-83fd-b1ac6521953c.png)
> The files should locate at %appdata%\flowlauncher\settings\setting.json.

I also think it may be the settings' action keywords @taooceros. @dcavacec can you try those intranet links with a fresh set of settings of flow please- from flow type `user data`, rename the containing folder, and flow will regenerate a fresh set of settings.

**Reply by ghost** [reply]
Does that mean renaming the "FlowLauncher" folder?

![image](https://user-images.githubusercontent.com/69704029/133023423-32ed17c8-b72a-461b-8e06-424e9e3e7191.png)
_Result of "user data" submitted from flow.launcher directly_

**Reply by jjw24** [reply]
Yep, exit flow and rename the folder 'FlowLauncher' to 'FlowLauncher-backup'. Run flow and you should see a new folder 'FlowLauncher' is created, this will contain the default settings. Add your site changes and see if it works.

To revert back to your old settings, delete the newly created 'FlowLauncher' folder and rename 'FlowLauncher-backup' back to 'FlowLauncher'.

**Reply by taooceros** [reply]
> > No error is presenting in the log.....It works on my computer with the same link🤔
> 
> What do  you mean by this? Guru and Hubble are internal to my workplace. Do you work where I work?

I was able to create the same link and open it in the browser.

**Reply by taooceros** [reply]
> Adding the actionKeywords "guru" and "hub" in _Settings.json_ didn't help, unfortunately. 
> 

Forget to mention, try closing flow before adding them?

**Reply by ghost** [reply]
I was able to fix the problem by closing Flow before editing Settings.json, then launching again. Thanks all for your input.

**Comment by ghost** [comment]
@deefrawley, gotcha. Yes, I was able to add amazon.com, which works:

![image](https://user-images.githubusercontent.com/69704029/132686791-e1e56a42-fea3-4cf0-981a-23e199c84f3f.png)

![image](https://user-images.githubusercontent.com/69704029/132686803-257db4f8-4902-45ad-abed-2278c22ca625.png)

**Reply by deefrawley** [reply]
Strange. When you said in your first post they "never appear" does this mean you type 'hub' or 'guru' and you don't see the magnifying glass and "Search Guru" in Flow?

**Reply by ghost** [reply]
> Strange. When you said in your first post they "never appear" does this mean you type 'hub' or 'guru' and you don't see the magnifying glass and "Search Guru" in Flow?

Correct

**Comment by ghost** [comment]
Something else I just realized. When attempting the workaround from [this post](https://github.com/Flow-Launcher/Flow.Launcher/discussions/686#discussioncomment-1299041), after restarting flow.launcher and going back into settings.json, "hub" and "guru" are gone.

**Reply by taooceros** [reply]
Sorry I forget to mention, you may need to close flow bofore editing the setting, or else it will be overwrited by the running flow (which won't change anything).

**Reply by ghost** [reply]
> Sorry I forget to mention, you may need to close flow bofore editing the setting, or else it will be overwrited by the running flow (which won't change anything).

This worked. The keywords appeared and launched the sites with the search results after updating the settings while Flow was closed. Thank you.
